hack/run-operator-tests.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+INTEGRATION_OUTPUT_JUNIT=${INTEGRATION_OUTPUT_JUNIT:-false}
+KUDO_VERSION=${KUDO_VERSION:-test}
+
+docker build . \
```

Comment:
why is there a docker build in this script?  shouldn't this be part of the precondition of this script?  don't we already have a build for docker in `make`?    
`docker build` is now in 3+ locations now.. can we consolidate?

Reply:
perhaps this docker build is slightly different... where it includes run-test-* scripts.  making a need for a different build.
there is a lot of commonality between the run-test scripts and even in 1 script there is duplication that could be cleaned up.   It is hard to see if consolidation will make things better or not... I would hate jumping from file to file to track down something simple... today that could mean 1. circle-ci file, 2. makefile, 3. run-script file 

The line `sed "s/%version%/$KUDO_VERSION/" kudo-e2e-test.yaml.tmpl > kudo-e2e-test.yaml` why it has different names across these files... it doesn't need to and they all do the same thing.   

I don't see any reason to change the file name to match "operator" tests vs e2e.   "operator" tests is another grouping of e2e tests with a different purpose is what I imagine without more clarification.

Reply:
This has to do with the test setup that is working around limitations in CircleCI. `run-tests.sh` starts a Docker container that can run Docker-in-Docker. This container then runs a test script in `hack/`, e.g. `hack/run-operator-tests.sh` in this case. The script builds a Docker image of the controller containing the changes of the PR under test and injects this into the `kind` cluster.
I don't like this complicated test setup and tried to refactor it to be simpler but ran into limitations in CircleCI: We have to run the tests in a VM environment to be able to start privileged containers in this VM. But we also need to install prerequisites for our tests in this VM and the easiest way is trough Docker, using the `test/Dockerfile`. That's why we have this complicated Docker-in-Docker setup here.